tests/ui/explicit_counter_loop.rs

```diff
@@ -113,3 +113,12 @@ mod issue_3308 {
         }
     }
 }
+
+mod issue_1670 {
+    pub fn test() {
+        let mut count = 0;
+        for _i in 3..10 {
+            count += 1;
```

Comment:
I think that was tested some times before, but could you also add a test, where the count variable is used after the loop? If that's the case the suggestion can't be applied.
```rust
let mut count = 0;
for _ in 3..10 {
    count += 1;
}
println!("{}", count);
```

Reply:
Nvm, this is tested here
https://github.com/rust-lang/rust-clippy/blob/2b82c71b559d9886e3ab3a7ad0194a5fe4aab58b/tests/ui/explicit_counter_loop.rs#L20-L25